### PR TITLE
Add manual traits

### PR DESCRIPTION
**Pull request by EPashkin**
Part of https://github.com/gtk-rs/gir/issues/683

currently only support traits from current crate
and don't use parents manual traits at all (so it need be added manually)

### CODE REVIEW

**Comment by EPashkin** [comment]
This is final version, support using parent traits and traits from other crates. All changes in configs linked to issue except gdk (it have only one manual trait)

@GuillaumeGomez, @sdroege Please take look when you have time.

**Comment by GuillaumeGomez** [comment]
Looks good to me, thanks!

**Comment by EPashkin** [comment]
Lets merge this too

File: README.md [hunk 1]
```diff
@@ -402,6 +402,21 @@ It'll generate a markdown file if everything went fine. That's where all this cr
...
+name = "Gtk.Assistant"
+status = "generate"
+#add link to trait from current crate
+manual_traits = ["AssistantExtManual"]
+
+[[object]]
+name = "Gtk.Application"
+status = "generate"
+#add link to trait from other crate
+manual_traits = ["gio::ApplicationExtManual"]
```
Comment (sdroege) [inline README.md]: I don't like this very much but don't have a better idea. Ideally we should be able to re-use the information from the `Gir.toml` of `gio` here but we don't have that...

File: README.md [hunk 2]
```diff
@@ -402,6 +402,21 @@ It'll generate a markdown file if everything went fine. That's where all this cr
 
 And now your crate should be completely documented as expected!
 
+If you defining traits manually you can add them to "Implements" section for classes and interfaces:
+```toml
+[[object]]
+name = "Gtk.Assistant"
+status = "generate"
+#add link to trait from current crate
+manual_traits = ["AssistantExtManual"]
```
Comment (sdroege) [inline README.md]: :+1:

**Pull request merged**